ansys/mapdl/core/mapdl.py

```diff
@@ -2173,6 +2173,15 @@ def run(self, command, write_to_log=True, **kwargs):
         >>> mapdl.prep7()
 
         """
+        if 'mute' in kwargs:
+            mute = kwargs['mute']
+        elif hasattr(self, 'mute'):
+            kwargs['mute'] = self.mute
+        else:
+            kwargs.setdefault("mute", False)
+
+        mute = kwargs['mute']
```

Comment:
This seems overly complicated. Why not include `mute=None` within the method signature of `run`.

Reply:
True. Good idea.

Reply:
I could do:

```py
if kwargs.get('mute') or self.mute if hasattr(self, mute) else False:
     mute = True
else:
     mute = False
```

I believe I need to check the class has  `mute` since it is specific of the grpc methods.

Reply:
we can't make "mute=False" part of the signature without modifying:

https://github.com/pyansys/pymapdl/blob/413c6f4a40171701d9c6ba09817ff9a91dd648cc/ansys/mapdl/core/mapdl_grpc.py#L608-L609

Reply:
Good point, because the other classes don't have the `mute` attribute.

I'd do this slightly differently to make it more readable:

```py
def run(..., run=None, **kwargs):

        if mute is None:
            if hasattr(self, 'mute'):  # if not gRPC
                mute = self.mute
            else:
                mute = False

        if mute:
            return
        ireturn response

```

Of course, we need to add documentation to the `run` method, which is missing and would be helpful to add in the first place.

Reply:
Seems good to me.

We are also passing the `mute` variable to `mapdl._run`:
```py

  text = self._run(command, mute=mute, **kwargs)

  if mute:
      return
```
Because of `MapdlCorba._run` and `MapdlConsole._run` do have `**kwargs` argument, the `mute` argument is stored there and do not raise an exception.